### PR TITLE
Rename some crates for publishing to crates.io

### PR DESCRIPTION
**Pull request by joao-paulo-parity**
For context see https://forum.parity.io/t/new-names-suggestions-for-squatted-crates/1513.

close https://github.com/paritytech/release-engineering/issues/132

polkadot companion: https://github.com/paritytech/polkadot/pull/6385
cumulus companion: https://github.com/paritytech/cumulus/pull/1944

### CODE REVIEW

File: utils/frame/remote-externalities/Cargo.toml
Comment (bkchr) [inline utils/frame/remote-externalities/Cargo.toml:2]:
```suggestion
name = "frame-remote-ext"
```

Or something similar, but it should be more expressive.

CC @kianenigma

**Review by bkchr — APPROVED** [review]
Besides the one nitpick, it looks good.

File: utils/frame/remote-externalities/Cargo.toml
Comment (joao-paulo-parity) [reply utils/frame/remote-externalities/Cargo.toml:2]: The other name I've reserved is `frame-remote-externalities`. I'll change to that, then.

**Comment by joao-paulo-parity** [comment]
I think the fix was effective as per https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2116743#L1811 which didn't fail to compile `polkadot-node-subsystem-test-helpers` like https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2115137#L1544. Now it's failing with a different error which appears to be unrelated to this PR:

```
Checking cumulus-relay-chain-inprocess-interface v0.1.0 (/builds/parity/mirrors/substrate/companions/cumulus/client/relay-chain-inprocess-interface)
error[E0308]: mismatched types
  --> client/relay-chain-minimal-node/src/network.rs:75:4
   |
75 | /             Some(Box::new(move |fut| {
76 | |                 spawn_handle.spawn("libp2p-node", Some("networking"), fut);
77 | |             }))
   | |_______________^ expected struct `Box`, found enum `std::option::Option`
   |
   = note: expected struct `Box<(dyn std::ops::Fn(Pin<Box<(dyn futures::Future<Output = ()> + std::marker::Send + 'static)>>) + std::marker::Send + 'static)>`
                found enum `std::option::Option<Box<[closure@client/relay-chain-minimal-node/src/network.rs:75:18: 75:28]>>`
```

I'll try to fix that.

**Comment by joao-paulo-parity** [comment]
The "different error" I mentioned in https://github.com/paritytech/substrate/pull/12837#issuecomment-1337065158 is very likely unrelated to this PR. I found a fix for it in https://github.com/paritytech/cumulus/pull/1917/commits/747400a0d7876a605290103b5d4341072b4e0868, i.e. it's related to https://github.com/paritytech/substrate/pull/12646.

File: client/beefy/Cargo.toml
Comment (acatangiu) [inline client/beefy/Cargo.toml:22]: nit:

since we're already touching tomls and also have companions for both Polkadot and Cumulus in this PR, why not go full rename from the get-go?

```suggestion
sp-beefy = { version = "4.0.0-dev", path = "../../primitives/beefy" }
```

**Review by acatangiu — APPROVED** [review]

**Review by lexnv — APPROVED** [review]

File: client/beefy/Cargo.toml
Comment (joao-paulo-parity) [reply client/beefy/Cargo.toml:22]: That would incur touching the `.rs` files as well, right? I followed @athei 's suggestion from https://forum.parity.io/t/new-names-suggestions-for-squatted-crates/1513/2 of only changing the `Cargo.toml`s.

I went with this approach because it minimizes the amount of changes I have to do in this PR and the companions. However, I'm open to whatever approach is the most agreed upon. Changing the `.rs` files wouldn't be difficult, it'd just be more work. Also it might require more reviews from other teams.

Do you strongly prefer the full rename or is it acceptable to leave it like this?

Comment (acatangiu) [reply client/beefy/Cargo.toml:22]: Unless there's some hurry with this PR, I would prefer the full rename, otherwise you can kick the can down the road and someone will do it later; not blocking the PR on this :)

Comment (ggwpez) [reply client/beefy/Cargo.toml:22]: I think we *are* a bit in a hurry with it; since we want to release the crates ASAP.

**Review by athei — APPROVED** [review]

**Pull request merged**
